source/backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -91,14 +93,34 @@ public PimsActivityInstance Add(PimsActivityInstance instance)
             return instance;
         }
 
-        public PimsActivityInstance Update(PimsActivityInstance instance)
+        public Tuple<PimsActivityInstance, PimsActivityInstanceNote> Update(PimsActivityInstance instance)
         {
             instance.ThrowIfNull(nameof(instance));
             var currentActivity = this.Context.PimsActivityInstances
                 .FirstOrDefault(x => x.ActivityInstanceId == instance.Id) ?? throw new KeyNotFoundException();
+
+            PimsActivityInstanceNote note = null;
+            if (currentActivity.ActivityInstanceStatusTypeCode!=instance.ActivityInstanceStatusTypeCode && this.User.HasPermission(Permissions.NoteAdd))
+            {
+                // add note
+                note = new PimsActivityInstanceNote()
```

Comment:
as stated above, the note should not be created here. In fact, I don't understand why the note is being created here since it is not being saved to the db. It looks like is is created here just so it can be passed back to the service, but why not just create it in the service instead?

Reply:
As the decision to create a note depends on earlier status and status changed by the user which is only known by the repository and not service.

Reply:
@Maheshbabu28 as an alternative you could implement a method in the repository that validates if an activity status is about to be changed. That way the service could call it and also be responsible of deciding if a note is required. You can see this pattern being used in **AcquisitionFileService** and repository where whenever a user tries to change the region of an acquisition file we ask the user for confirmation first (via an exception if the user didn't confirm). 

In your case you wouldn't be throwing an exception but instead creating a note if the repo tells the service that the status of the incoming activity doesn't match the one stored in the DB

Reply:
Thanks @asanchezr  for detailed explanation, I have made the changes.

Reply:
@Maheshbabu28 can we see those changes? I don't see them here.

Reply:
@devinleighsmith https://github.com/bcgov/PSP/pull/2223/commits/20efbcb1d9e8698cda64117683ef00164f054114